### PR TITLE
Only raise a warning if subscription is not enabled.

### PR DESCRIPTION
**Pull request by djberg96**
Upon further review, it seems better to emit a warning, rather than raise an error, if a disabled subscription is found. You can still perform read operations, even with a disabled subscription.

### CODE REVIEW

**Comment by djberg96** [comment]
@bronaghs @bzwei Please review.

File: lib/azure/armrest/armrest_service.rb [hunk 1]
```diff
@@ -137,11 +137,15 @@ def self.fetch_subscription_id(config)
           :authorization => config.token
         )
 
-        subscriptions = JSON.parse(response)["value"]
+        hash = JSON.parse(response)["value"].first
```
Comment (bzwei) [inline lib/azure/armrest/armrest_service.rb]: Shouldn't you still need to find the first enabled subscription? Use and warn the first subscription only if no enabled subscription is found. Raise error if no subscription is found at all.

**Comment by djberg96** [comment]
@bzwei Ok, updated the PR to prefer enabled subscriptions first.

File: lib/azure/armrest/armrest_service.rb [hunk 2]
```diff
@@ -137,11 +137,18 @@ def self.fetch_subscription_id(config)
...
+
+        # Look for the first enabled subscription, otherwise just take the first subscription found.
+        hash = array.find{ |h| h['state'] == 'Enabled' } || array.first
+
+        id = hash.fetch('subscriptionId')
 
-        raise ArgumentError, "No enabled or associated subscription found" if hash.nil?
+        warn "Warning: subscription #{id} is not enabled" unless hash['state'] == 'Enabled'
 
         @@subscriptions[config.as_cache_key] = hash.fetch('subscriptionId')
```
Comment (bzwei) [inline lib/azure/armrest/armrest_service.rb]: `@@subscription[config.as_cache_key] = id`

Comment (djberg96) [inline lib/azure/armrest/armrest_service.rb]: Fixed. Forgot to change it back.

**Comment by djberg96** [comment]
@bzwei The rebuild worked this time. :)

**Pull request merged**